### ISSUE TITLE
Eloquent\Collection::map() returns Eloquent\Collection even when callback produces non-Model values

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#500]
## Description

When `Eloquent\Collection::map()` transforms models into arrays (or any non-Model type), Psalm infers the return type as `Eloquent\Collection<int, array{...}>`. This violates the `TModel of Model` template constraint on `Eloquent\Collection`, causing `InvalidTemplateParam` errors.

Laravel's actual runtime behavior: `map()` checks if the result contains non-Model values and calls `toBase()` to return a `Support\Collection` instead. The stub should reflect this.

## Impact

**89 `InvalidTemplateParam`** + related `ArgumentTypeCoercion` false positives. Found while auditing [monicahq/monica](https://github.com/monicahq/monica) with plugin v4.2.0 + Psalm 7.

## Reproduction

```php
// Any Eloquent Collection mapped to arrays
$users = User::all(); // Eloquent\Collection<int, User>

$names = $users->map(fn (User $u) => ['name' => $u->name]);
// Psalm infers: Eloquent\Collection<int, array{name: string}>
// InvalidTemplateParam: array{name: string} does not satisfy TModel of Model
//
// Laravel actually returns: Support\Collection<int, array{name: string}>
```

## Fix

Add `map()` and `mapWithKeys()` overrides to the `Eloquent\Collection` stub with conditional return types:

```php
/**
 * @template TMapValue
 * @param  callable(TModel, TKey): TMapValue  $callback
 * @psalm-return (TMapValue is \Illuminate\Database\Eloquent\Model ? static<TKey, TMapValue> : \Illuminate\Support\Collection<TKey, TMapValue>)
 */
public function map(callable $callback) {}
```

This mirrors Laravel's runtime behavior: returns `static` (Eloquent\Collection) when all mapped values are Model instances, falls back to `Support\Collection` otherwise.

## Related

- #514 — TRelatedModel lost through `__call` forwarding (different root cause, overlapping symptoms)
